src/common/AndroidUtils.ts

```diff
@@ -333,61 +322,66 @@ export class AndroidSDKUtils {
         device: string,
         abi: string
     ): Promise<boolean> {
-        const createAvdCommand = `${AndroidSDKUtils.getAvdManagerCommand()} create avd -n ${emulatorName} --force -k ${AndroidSDKUtils.systemImagePath(
+        // Just like Android Studio AVD Manager GUI interface, replace blank spaces with _ so that the ID of this AVD
+        // doesn't have blanks (since that's not allowed). AVD Manager will automatially replace _ back with blank
+        // to generate user friendly display names.
+        const resolvedName = emulatorName.replace(/ /gi, '_');
+
+        const createAvdCommand = `${AndroidSDKUtils.getAvdManagerCommand()} create avd -n '${resolvedName}' --force -k ${AndroidSDKUtils.systemImagePath(
             platformAPI,
             emulatorImage,
             abi
         )} --device ${device} --abi ${abi}`;
-        return new Promise((resolve, reject) => {
-            try {
-                const child = AndroidSDKUtils.spawnChild(createAvdCommand);
-                child.stdin.setDefaultEncoding('utf8');
-                child.stdin.write('no');
-                if (child) {
-                    child.stdout.on('data', () => {
-                        setTimeout(() => {
-                            resolve(true);
-                        }, 3000);
-                    });
-                    child.stdout.on('exit', () => resolve(true));
-                    child.stderr.on('error', () => reject(false));
-                } else {
-                    reject(false);
-                }
-            } catch (error) {
-                reject(error);
-            }
-        }).then((resolve) =>
-            AndroidSDKUtils.updateEmulatorConfig(emulatorName)
-        );
+
```

Comment:
I'm not sure why we were using `spawnChild` previously.... but it was problematic. If the command for the spawned process failed for some reason, the callback was not being invoked and we would proceed as if it had succeeded. So even if the create command had failed, we would report it as succeeded.

I switched to using (synchronous) `executeCommand` and it seems to be working fine. I don't know if there will be issues with much older versions of Android Studio and whether that was the reason for using `spawnChild` previously.

I'll try to spend some time installing older versions of AS and see if this change still works with it.

Reply:
Even on older versions of Android Studio as far back as 2018, this new approach seems to be working fine.